### DISCUSSION TITLE
【求助】大神，像这种分左右两栏的网页不能紧挨着拼接是怎么回事？

### DISCUSSION THREAD

**Discussion by ahwgit** [hoothin/UserScripts#434] [Q&A]
先给个网址https://www.zhaojiafang.com/shop/index.php?act=show_store&op=goods_all&store_id=773，
像这种分左右两栏的网页，我发现，左侧栏内容比较短时，右侧栏分页内容能够紧挨着拼接；如果左侧栏内容很长时，右侧栏分页内容就不能紧挨着拼接了？要拉到底才有拼接页，非常不方便！能否改进代码？
上面给出的网页地址，在我的电脑上是不能紧挨着拼接的。

**Comment by hoothin** [comment]
遇到这种异形结构的网页，就自己编辑规则啊。我不是写了`主体框架抓取器`了吗？你倒是用起来啊
```
{
    "name": "找家纺",
    "url": "^https?://www\\.zhaojiafang\\.com/shop/",
    "example": "https://www.zhaojiafang.com/shop/index.php?act=show_store&op=goods_all&store_id=773%EF%BC%8C",
    "pageElement": "div.content.ncs-all-goods-list",
}
```

**Reply by ahwgit** [reply]
不太会写，，非常感谢！

**Comment by hoothin** [comment]
以“找家纺”这个站为例，编辑规则是纯鼠标操作，不需要额外的知识。具体步骤如下，你自己过一下就能明白
1. 点击“编辑此站规则”
2. 移动鼠标到商品列表上，使得蓝框正好覆盖列表区域，点击以选定区域
3. 保存规则

**Reply by ahwgit** [reply]
感谢！问题已解决

**Reply by ahwgit** [reply]
大神，今天将永页机更新至v.1.9.36.91，发现“找家纺”的自定义规则没什么作用了，只加载第一页和最后一页，太奇怪了。于是我把“找家纺”的自定义规则删除之后，再刷新页面，还是只加载第一页和最后一页！不知道哪里出了问题？其他网站都能正常拼接

**Reply by hoothin** [reply]
旧版本是正常的吗？这个是末页识别成下一页了，用`"nextLink":".active+li>a"`就能解决

**Reply by ahwgit** [reply]
旧版本是正常的，当时想返回上一个版本，没有旧版本可返，现在更新到92貌似正常了，谢谢。我再看看其他网站情况怎么样

**Reply by ahwgit** [reply]
在吾爱破解论坛里搜索，比如office tab，拼接页是从page1开始，而不是page2

**Reply by hoothin** [reply]
这是正常的，page 显示的数字是根据 url 变化的，例如你从 79 页翻页，那下一页显示的就是 80 页。
百度站内搜索这边第二页的 url 里有个 p=1，因为它自身就是从 1 开始数的，所以永页机显示的也是 page1

**Reply by ahwgit** [reply]
明白了，谢谢
